scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -605,6 +613,49 @@ s32 BPF_STRUCT_OPS(rusty_select_cpu, struct task_struct *p, s32 prev_cpu,
 			}
 		}
 	}
+	
+	// the task might also be forked/execd so check for wake up only
+	if (wake_flags & SCX_WAKE_TTWU) {
+		cpu = scx_bpf_pick_idle_cpu((const struct cpumask *)p_cpumask, 0);
+		if (cpu >= 0) {
+			goto direct;
+		}
+		if (scx_bpf_dsq_nr_queued(taskc->dom_id) == 0) {
```

Comment:
Hmm... what if the task being queued is higher priority than others which are already queued?

Reply:
That is for my TODO below.

Reply:
So, if you track the vtime of the current task, this likely will be solved at the same time.

Reply:
Not sure I understand how it solves both. Don't I still need to maintain a list of queued tasks for each CPU so I can do the vtime comparison?

Reply:
You know that the task which is currently executing had the earliest vtime when it started executing and can also calculate the current vtime of the task from the execution time delta and its weight. Now, this doesn't guraantee that the next task in queue is after the current vtime as we can overrun that within slice. We can probably add a peeking operation to dsq to make it water tight but for now just comparing against the current executing task's vtime should do. Note that we can be more conservative to avoid being wrong - ie. preempt iff the preemptor's vtime is before the preemptee's vtime at the time it started execution.

Reply:
So I think the latest updates reflect this suggestion (particularly the last sentence where we look at (store) the vtime of the task when it starts running.